### PR TITLE
[Review] project redesign

### PR DESCRIPTION
**Pull request by phongyanon**
please review and verify to improve

### CODE REVIEW

File: specific/cmo_purchase/models/sale.py [hunk 1]
```diff
@@ -2,18 +2,52 @@
...
+        return super(SaleOrderLine, self).name_search(
+            name, args=args, operator=operator, limit=limit)
+
+
+class SaleLayoutCategory(models.Model):
+    _inherit = 'sale_layout.category'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        context = self._context.copy()
```
Comment (kittiu) [inline specific/cmo_purchase/models/sale.py:44]: Actually, you don't need this line (we are not changing any context value, right?)

File: specific/cmo_purchase/models/sale.py [hunk 2]
```diff
@@ -2,18 +2,52 @@
...
+#             name, args=args, operator=operator, limit=limit)
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
 
     @api.model
     def name_search(self, name, args=None, operator='ilike', limit=100):
         context = self._context.copy()
+        # filter order line from quotation
```
Comment (kittiu) [inline specific/cmo_purchase/models/sale.py:28]: Actually, you don't need this line (we are not changing any context value, right?)
context = self._context.copy()

File: specific/cmo_purchase/models/sale.py (hunk 1 above)
Comment (kittiu) [inline specific/cmo_purchase/models/sale.py:44]: We don't need this line. We are not changing context value.
context = self._context.copy()

Comment (phongyanon) [reply specific/cmo_purchase/models/sale.py:44]: yes we 're not change any context

**Pull request merged**